test_data/asm/trait_parsing.asm

```diff
@@ -0,0 +1,13 @@
+machine Trait with degree: 8 {
+    reg pc[@pc];
+    reg X[<=];
+    reg A;
+
+    trait ArraySum<T> {
+        array_sum: T[4 + 1] -> T,
```

Comment:
To check that the path canonicalizer is correctly implemented, could you add a generic enum type and use it here like for example `Option<T>`?